packages/modal-ui/src/lib/components/DerivationPath.tsx

```diff
@@ -83,54 +68,55 @@ export const DerivationPath: React.FC<DerivationPathProps> = ({
     const accountIds = await response.json();
 
     if (!Array.isArray(accountIds) || !accountIds.length) {
-      throw new Error(
-        "Failed to find account linked for public key: " + publicKey
-      );
+      return [];
     }
 
     return accountIds;
   };
 
-  const resolveAccounts = async (wallet: Wallet) => {
-    const accounts: Array<HardwareWalletAccountState> = [];
-
-    for (let i = 0; i < derivationPaths.length; i += 1) {
-      const derivationPath = derivationPaths[i].path;
-
-      if (wallet.type === "hardware") {
-        const publicKey = await wallet.getPublicKey(derivationPath);
-        const accountIds = await getAccountIdsFromPublicKey(publicKey);
-
-        accounts.push({
-          derivationPath,
-          publicKey,
-          accountIds,
-          selectedAccountId: accountIds[0],
-        });
-      }
+  const resolveAccount = async (
```

Comment:
Probably just a naming thing but this is really `resolveAccounts` and should maybe return the list of accounts from `getAccountIds`?

Reply:
In fact this abstraction causes problems in `handleValidateAccount`  as we should only display the custom input screen if we get an error from `getAccountIds` and not when the user rejects the `getPublicKey` request. No accounts should display a screen similar to Mutisafe